File: embedded_NodeMcu/poker_lib/Hand.cpp
#include "Hand.h"

Hand::Hand(byte card_data_1, byte card_data_2)
{
  _card_data_1 = card_data_1;
  _card_data_2 = card_data_2;
}

Card Hand::card1(){
  return Card(_card_data_1);
}

Card Hand::card2(){
  return Card(_card_data_2);
}

String Hand::toString(){
  return "[" + card1().toString() + " | " + card2().toString() + "]";
}

File: dotnet_testnodes/Poker.Shared/CardsLogic/Hand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Poker.Shared.CardsLogic
{
    public class Hand
    {
        public Card Card1 { get; set; }
        public Card Card2 { get; set; }

        public Hand(Card card1, Card card2)
        {
            if (card1.Value == card2.Value && card1.Color == card2.Color)
                throw new ArgumentException("Cards cannot be identical. Value " + card1.Value + " color " + card1.Color);
            Card1 = card1 ?? throw new ArgumentNullException(nameof(card1));
            Card2 = card2 ?? throw new ArgumentNullException(nameof(card2));
        }

        public override string ToString()
        {
            return Card1 + ", " + Card2;
        }
    }
}
